nautobot/ipam/tests/test_models.py

```diff
@@ -962,6 +962,78 @@ def setUp(self):
+        # Use a namespace different from the default namespace which has an ipaddress
+        namespace = (
+            Namespace.objects.exclude(pk=default_namespace.pk).filter(prefixes__ip_addresses__isnull=False).first()
+        )
+        # Creating a new status; because we require a unique object when doing a
+        # `get_or_create` without passing in `host/mask_length`` to avoid receiving multiple IP addresses
+        status = Status.objects.create(name="Test Status 1")
+        ipaddress = IPAddress.objects.filter(parent__namespace=namespace).first()
+        host = ipaddress.host
+        mask_length = ipaddress.mask_length
+
+        ipaddress.status = status
+        ipaddress.save()
+
+        with self.subTest("Assert retrieve"):
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host,
+                mask_length=mask_length,
+                namespace=namespace,
+                status=status,
+            )
+            self.assertEqual(ip_obj, ipaddress)
+            self.assertFalse(created)
+
+        with self.subTest(
+            "Assert get_or_create utilizes default namespace when retrieving parent if no namespace is provided"
+        ):
+            ipaddress.parent.namespace = default_namespace
+            ipaddress.parent.save()
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host,
+                mask_length=mask_length,
+                status=status,
+            )
+            self.assertEqual(ip_obj, ipaddress)
+            self.assertFalse(created)
+
+        with self.subTest("Assert create"):
+            ipaddress.delete()
+
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host,
+                mask_length=mask_length,
+                status=status,
+            )
+            self.assertEqual(ip_obj.host, host)
+            self.assertEqual(ip_obj.mask_length, mask_length)
+            self.assertEqual(ip_obj.parent.namespace, default_namespace)
+            self.assertTrue(created)
+
+        with self.subTest("Assert create explicitly defining a non default namespace"):
+            ip_obj.delete()
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host, mask_length=mask_length, status=status, namespace=namespace
+            )
+            expected_parent_for_namespace = Prefix.objects.filter(namespace=namespace).get_closest_parent(
+                cidr=f"{host}/{mask_length}", include_self=True
+            )
+            self.assertEqual(ip_obj.host, host)
+            self.assertEqual(ip_obj.mask_length, mask_length)
+            self.assertEqual(ip_obj.parent, expected_parent_for_namespace)
+            self.assertEqual(ip_obj.parent.namespace, namespace)
+            self.assertTrue(created)
+
+        with self.subTest("Assert if host/mask_length is not present"):
+            ip_obj, created = IPAddress.objects.get_or_create(status=status)
```

Comment:
Kind of a strange test if you ask me... if you're really wanting to test `get` with partial information, I'd think that specifying the `host` without a `mask_length` might be more realistic.